src/tables.jl

```diff
@@ -6,3 +6,30 @@ Tables.columnaccess(::Type{<:StructVector}) = true
 Tables.columns(s::StructVector) = fieldarrays(s)
 
 Tables.schema(s::StructVector) = Tables.Schema(staticschema(eltype(s)))
+
+function Base.append!(s::StructVector, rows)
+    if Tables.isrowtable(rows) && Tables.columnaccess(rows)
+        # Input `rows` is a container of rows _and_ satisfies column
+        # table interface.  Thus, we can add the input column-by-column.
+        table = Tables.columns(rows)
+        isempty(_setdiff(propertynames(s), Tables.columnnames(rows))) ||
+            _invalid_columns_error(s, rows)
+        _foreach(propertynames(s)) do name
```

Comment:
Generally for this StructArrays uses `foreachfield`, but that assumes that also `rows` implements `getproperty` for column access, which in the latest Tables doesn't need to be the case. What I don't understand is: how is `foldl` better than `foreach` for tuples? Is it is as efficient as the generated function?

Reply:
By looking at the `@code_llvm` or `@code_native` for `mapfoldl` on tuples, it seems to generate the optimal thing, I'm just a bit confused as to where is the special method for tuples.

Reply:
Thanks for merging this. Maybe you've already figured it out why, but the reason why I didn't use `foreachfield` there was that AFAICT `foreachfield` does not pass property names to the function.